cram_pr2/cram_pr2_shopping_demo/src/plans.lisp

```diff
@@ -0,0 +1,214 @@
+                               "map" 0.0
+                               (cl-transforms:make-3d-vector -1 -0.35 0)
+                               (cl-transforms:axis-angle->quaternion
+                                (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+
+(defparameter *pose-grasping-right* (cl-transforms-stamped:make-pose-stamped
+                               "map" 0.0
+                               (cl-transforms:make-3d-vector -2 -0.35 0)
+                               (cl-transforms:axis-angle->quaternion
+                                (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+
+
+(defparameter *pose-near-table* (cl-transforms-stamped:make-pose-stamped
+                                  "map" 0.0
+                                  (cl-transforms:make-3d-vector -2.5 0.5 0)
+                                  (cl-transforms:axis-angle->quaternion
+                                   (cl-transforms:make-3d-vector 0 0 1) (/ pi 1))))
+
+(defparameter *looking-pose-mid* (cl-transforms-stamped:make-pose-stamped
+                                  "map" 0.0
+                                  (cl-transforms:make-3d-vector -1.5 -1.05 1.15)
+                                  (cl-transforms:make-identity-rotation)))
+
+(defparameter *pose-detecting* (cl-transforms-stamped:make-pose-stamped
+                               "map" 0.0
+                               (cl-transforms:make-3d-vector -1.5 1 0)
+                               (cl-transforms:axis-angle->quaternion
+                                (cl-transforms:make-3d-vector 0 0 1) (/ pi -2))))
+
+(defun move-object (?object)
+    (let ((?grasping-arm :right)
+          (?grasp-pose *pose-grasping*)
+          (?grasp-pose-left *pose-grasping-left*)
+          (?grasp-pose-right *pose-grasping-right*)
+          (?pose-near-table *pose-near-table*)
+          (?table *table*)
+          (map->obj)
+          (map->base)
+          (trans)
+          (?newpose)
+          (?newtransform))
+
+      (exe:perform 
+       (desig:a motion
+                (type moving-torso)
+                (joint-angle 0)))
+      
+      (exe:perform
+       (desig:an action
+                 (type going)
+                 (target (desig:a location
+                                  (pose ?grasp-pose)))))
+      
+
+      (exe:perform
+       (desig:an action
+                 (type positioning-arm)
+                 (left-configuration park)
+                 (right-configuration park)))
+      
+
+      ;; creating the new Transform from the base_footprint infront of the shelf and the object
+      (setf map->base (cl-transforms:make-transform
+                       (cl-transforms:translation (cl-transforms:reference-transform *pose-grasping*))
+                       (cl-transforms:orientation *pose-grasping*)))
+                       
+      
+      (setf map->obj (second (fourth (desig:desig-prop-value ?object :pose))))
```

Comment:
Don't use ``forth`` here, instead use ``find`` or similar to find the ``:pose-in-map`` key.